File: app/src/main/java/vasuki/istanpdf/domain/SaveDocx.java
package vasuki.istanpdf.domain;

import android.net.Uri;

import java.util.List;

import vasuki.istanpdf.data.DocxEngine;
import vasuki.istanpdf.model.PageItem;

public final class SaveDocx {
    private final DocxEngine docxEngine;

    public SaveDocx(DocxEngine docxEngine) {
        this.docxEngine = docxEngine;
    }

    public void execute(Uri source, List<PageItem> snapshot, Uri destination) throws Exception {
        if (source == null) {
            throw new IllegalArgumentException("Source DOCX is required");
        }
        docxEngine.saveEdited(source, snapshot, destination);
    }
}

File: app/src/main/kotlin/vasuki/istanpdf/domain/SaveDocx.kt
package vasuki.istanpdf.domain

import android.net.Uri
import vasuki.istanpdf.data.DocxEngine
import vasuki.istanpdf.model.PageItem

class SaveDocx(private val docxEngine: DocxEngine) {
    @Throws(Exception::class)
    fun execute(source: Uri, snapshot: List<PageItem>, destination: Uri) {
        docxEngine.saveEdited(source, snapshot, destination)
    }
}
